python/src/lib.rs

```diff
@@ -528,6 +531,40 @@ impl RawDeltaTable {
         Ok(serde_json::to_string(&metrics).unwrap())
     }
 
+    #[pyo3(signature = (fields, custom_metadata=None))]
+    pub fn add_column(
+        &mut self,
+        py: Python,
+        fields: Vec<Field>,
+        custom_metadata: Option<HashMap<String, String>>,
+    ) -> PyResult<()> {
+        let table = py.allow_threads(|| {
+            let mut cmd = AddColumnBuilder::new(
+                self._table.log_store(),
+                self._table.snapshot().map_err(PythonError::from)?.clone(),
+            );
+
+            let new_fields = fields
+                .iter()
+                .map(|v| v.inner.clone())
+                .collect::<Vec<StructField>>();
+
+            cmd = cmd.with_fields(new_fields);
```

Comment:
Since `fields` is owned, can this just consume the value? e.g.

```rust
cmd = cmd.with_fields(fields.into_iter().map(|v| v.inner)))
```

(or some such nonsense :shrug: )

Reply:
Hmm I'll have to try, reason I cloned is because StructField didn't have the Copy Trait